app/src/main/java/org/p2p/wallet/auth/interactor/MetadataInteractor.kt

```diff
@@ -31,30 +36,162 @@ class MetadataInteractor(
         }
     }
 
+    suspend fun updateMetadata(metadata: GatewayOnboardingMetadata) {
+        currentMetadata = metadata
```

Comment:
i would rather always fetch KEY_ONBOARDING_METADATA from storage than keep an var variable inside the interactor

you can also extract this storage to some sort of LocalRepository